docs/architecture/adr-020-protobuf-transaction-encoding.md

```diff
@@ -284,27 +285,21 @@ and `FileDescriptor`s and returns a boolean result.
 
 ### Public Key Encoding
 
-Public keys in the Cosmos SDK implement Tendermint's `crypto.PubKey` interface,
-so a natural solution might be to use `Any` as we are doing for other interfaces.
-There are, however, a limited number of public keys in existence and new ones
-aren't created overnight. The proposed solution is to use a `oneof` that:
-
-- attempts to catalog all known key types even if a given app can't use them all
-- has an `Any` member that can be used when a key type isn't present in the `oneof`
+Public keys in the Cosmos SDK implement Tendermint's `crypto.PubKey` interface.
+Following public keys are implemented: secp256k1, ed25519 and multisignature, which
+correspond to individual protobuf types.
 
 Ex:
 
 ```proto
-message PublicKey {
-    oneof sum {
-        bytes secp256k1 = 1;
-        bytes ed25519 = 2;
-        ...
-        google.protobuf.Any any_pubkey = 15;
-    }
+message PubKey {
+    bytes key = 1;
 }
 ```
 
+`multisig.LegacyAminoPubKey` has a slice of `Any`'s member to support any
+protobuf public key type.
```

Comment:
Could you reward it? `slice` can be confused here. How about:

> `multisig.LegacyAminoPubKey` contains an array `Any` to support any
protobuf public key type.